Detectors/HMPID/calibration/src/HMPIDDCSProcessor.cxx

```diff
@@ -0,0 +1,803 @@
+  }
+}
+
+// process Tempin
+void HMPIDDCSProcessor::finalizeTempIn(int iCh, int iRad)
+{
+
+  if (dpVecTempIn[3 * iCh + iRad].size() != 0) {
+    cntTin = 0;
+
+    auto minTime = getMinTime(dpVecTempIn[3 * iCh + iRad]);
+    auto maxTime = getMaxTime(dpVecTempIn[3 * iCh + iRad]);
+
+    std::unique_ptr<TGraph> pGrTIn;
+    pGrTIn.reset(new TGraph);
+
+    for (DPCOM dp : dpVecTempIn[3 * iCh + iRad]) {
+      auto dpVal = o2::dcs::getValue<double>(dp);
+      auto time = dp.data.get_epoch_time(); //-minTime
+      pGrTIn->SetPoint(cntTin++, time, dpVal);
+    }
+    // maxTime -= minTime;
+    // minTime = 0;
+    std::unique_ptr<TF1> pTin;
+    pTin.reset(
+      new TF1(Form("Tin%i%i", iCh, iRad), "[0]+[1]*x", minTime, maxTime));
+
+    if (cntTin == 1) {
+      pGrTIn->GetPoint(0, xP, yP);
+      pTin->SetParameter(0, yP);
+      pTin->SetParameter(1, 0);
+    } else {
+      pGrTIn->Fit(Form("Tin%i%i", iCh, iRad), "Q");
+    }
+
+    pTin->SetTitle(Form("Temp-In Fit Chamber%i Radiator%i; Time [ms]; Temp [C]",
+                        iCh, iRad));
+    std::vector<TF1>::iterator itTin = arNmean.begin() + 6 * iCh + 2 * iRad;
+    arNmean.insert(itTin, *(pTin.get()));
+
+  } else {
+    LOG(info) << Form("No entries in temp-in for ch%irad%i", iCh, iRad);
+  }
+}
+
+// returns nullptr if the element in array of DPCOM-vector has no entries
+std::unique_ptr<TF1> HMPIDDCSProcessor::finalizeHv(int iCh, int iSec)
+{
+  std::unique_ptr<TF1> pHvTF;
+  if (dpVecHV[3 * iCh + iSec].size() != 0) {
+
+    std::unique_ptr<TGraph> pGrHV;
+    pGrHV.reset(new TGraph);
+    cntHV = 0;
+
+    hvFirstTime = getMinTime(dpVecHV[3 * iCh + iSec]);
+    hvLastTime = getMaxTime(dpVecHV[3 * iCh + iSec]);
+
+    for (DPCOM dp : dpVecHV[3 * iCh + iSec]) {
+      auto dpVal = o2::dcs::getValue<double>(dp);
+      auto time = dp.data.get_epoch_time(); //- hvFirstTime
+      pGrHV->SetPoint(cntHV++, time, dpVal);
+    }
+    // hvLastTime -= hvFirstTime;
+    // hvFirstTime = 0;
+    if (cntHV == 1) {
+      pGrHV->GetPoint(0, xP, yP);
+      (pHvTF).reset(new TF1(Form("HV%i_%i", iCh, iSec), Form("%f", yP),
+                            hvFirstTime, hvLastTime));
+      pArrHv[3 * iCh + iSec] = *(pHvTF.get());
+    } else {
+      (pHvTF).reset(new TF1(Form("HV%i_%i", iCh, iSec), "[0]+x*[1]",
+                            hvFirstTime, hvLastTime));
+      pGrHV->Fit(Form("HV%i_%i", iCh, iSec), "Q");
+      pArrHv[3 * iCh + iSec] = *(pHvTF.get());
+    }
+
+    return pHvTF;
+  }
+  LOG(info) << Form("No entries in HV for ch%isec%i", iCh, iSec);
+  return pHvTF;
+}
+
+// Process all arrays of DPCOM-vectors, perform fits
+// Called At the end of the Stream of DPs, in
+// HMPIDDCSDataProcessorSpec::endOfStream() function
+void HMPIDDCSProcessor::finalize()
+{
+
+  std::unique_ptr<TF1> pEnv = finalizeEnvPressure();
+  for (int iCh = 0; iCh < 7; iCh++) {
+
+    std::unique_ptr<TF1> pChPres = finalizeChPressure(iCh);
```

Comment:
What is the role of these histos? They are created and deleted when going out of scope w/o being used

Reply:
Indeed, are not in the CCDB but used to create the objects (charge threshold) to be stored in the CCDB (see lines 717 and 718 ) and then deleted.

Reply:
I see that you are using the name only, why do you need to create a histo?

Reply:
They are not an histos, they are TF1 objects coming from the fit of histos.

Reply:
OK, still don't fully understand the logic, but given that it is done once per run at finalization, the overhead should not be important.